x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/response/cohere/CohereEmbeddingsResponseEntityTests.java

```diff
@@ -0,0 +1,434 @@
+                    }
+                },
+                "response_type": "embeddings_floats"
+            }
+            """;
+
+        TextEmbeddingResults parsedResults = CohereEmbeddingsResponseEntity.fromResponse(
+            mock(Request.class),
+            new HttpResult(mock(HttpResponse.class), responseJson.getBytes(StandardCharsets.UTF_8))
+        );
+
+        MatcherAssert.assertThat(
+            parsedResults.embeddings(),
+            is(
+                List.of(
+                    TextEmbeddingResults.Embedding.ofFloats(List.of(-0.0018434525F, 0.01777649F)),
+                    TextEmbeddingResults.Embedding.ofFloats(List.of(-0.123F, 0.123F))
+                )
+            )
+        );
+    }
+
+    public void testFromResponse_FailsWhenEmbeddingsFieldIsNotPresent() {
+        String responseJson = """
+            {
+                "id": "3198467e-399f-4d4a-aa2c-58af93bd6dc4",
+                "texts": [
+                    "hello"
+                ],
+                "embeddings_not_here": [
+                    [
+                        -0.0018434525,
+                        0.01777649
+                    ]
+                ],
+                "meta": {
+                    "api_version": {
+                        "version": "1"
+                    },
+                    "billed_units": {
+                        "input_tokens": 1
+                    }
+                },
+                "response_type": "embeddings_floats"
+            }
+            """;
+
+        var thrownException = expectThrows(
+            IllegalStateException.class,
+            () -> CohereEmbeddingsResponseEntity.fromResponse(
+                mock(Request.class),
+                new HttpResult(mock(HttpResponse.class), responseJson.getBytes(StandardCharsets.UTF_8))
+            )
+        );
+
+        MatcherAssert.assertThat(
+            thrownException.getMessage(),
+            is("Failed to find required field [embeddings] in Cohere embeddings response")
+        );
+    }
+
+    public void testFromResponse_FailsWhenEmbeddingsByteValue_IsOutsideByteRange_Negative() {
+        String responseJson = """
+            {
+                "id": "3198467e-399f-4d4a-aa2c-58af93bd6dc4",
+                "texts": [
+                    "hello"
+                ],
+                "embeddings": {
+                    "int8": [
+                        [
+                            -129,
+                            127
+                        ]
+                    ]
+                },
+                "meta": {
+                    "api_version": {
+                        "version": "1"
+                    },
+                    "billed_units": {
+                        "input_tokens": 1
+                    }
+                },
+                "response_type": "embeddings_floats"
+            }
+            """;
+
+        var thrownException = expectThrows(
+            IllegalArgumentException.class,
```

Comment:
It seems to me that these two types of exceptions are actually the same: cohere failed to respond in a way that we could process. I think they should be throwing the same type of exception, maybe a failed dependency exception. 

Maybe these different java exceptions are being translated into rest exceptions somewhere that I missed?

Reply:
Yeah good point. They do get translated into rest exceptions but it's just a generic 500 I believe. I'll address this in a follow up PR because it'd be good to do this for all the inference service implementations.